PROTO_tests/tests/numeric_test.py

```diff
@@ -0,0 +1,362 @@
+    (ak.int64, ak.int64),
+    (ak.int64, ak.float64),
+    (ak.int64, npstr),
+    (ak.float64, ak.float64),
+    (ak.float64, npstr),
+    (ak.uint8, ak.int64),
+    (ak.uint8, ak.float64),
+    (ak.uint8, npstr),
+]
+
+
+class TestNumeric:
+
+    @pytest.mark.parametrize("numeric_type", NUMERIC_TYPES)
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    def test_seeded_rng_typed(self, prob_size, numeric_type):
+        seed = pytest.seed if pytest.seed is not None else 8675309
+
+        # Make sure unseeded runs differ
+        a = ak.randint(0, 2**32, prob_size, dtype=numeric_type)
+        b = ak.randint(0, 2**32, prob_size, dtype=numeric_type)
+        assert not ((a == b).all())
+
+        # Make sure seeded results are same
+        a = ak.randint(0, 2**32, prob_size, dtype=numeric_type, seed=seed)
+        b = ak.randint(0, 2**32, prob_size, dtype=numeric_type, seed=seed)
+        assert a.to_list() == b.to_list()
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    def test_seeded_rng_general(self, prob_size):
+        seed = pytest.seed if pytest.seed is not None else 8675309
+        # Uniform
+        assert not (ak.uniform(prob_size) == ak.uniform(prob_size)).all()
+        assert (
+            ak.uniform(prob_size, seed=seed).to_list()
+            == ak.uniform(prob_size, seed=seed).to_list()
+        )
+
+        # Standard Normal
+        assert not (ak.standard_normal(prob_size) == ak.standard_normal(prob_size)).all()
+        assert (
+            ak.standard_normal(prob_size, seed=seed).to_list()
+            == ak.standard_normal(prob_size, seed=seed).to_list()
+        )
+
+        # Strings (uniformly distributed length)
+        assert not (
+            ak.random_strings_uniform(1, 10, prob_size)
+            == ak.random_strings_uniform(1, 10, prob_size)
+        ).all()
+        assert (
+            ak.random_strings_uniform(1, 10, prob_size, seed=seed).to_list()
+            == ak.random_strings_uniform(1, 10, prob_size, seed=seed).to_list()
+        )
+
+        # Strings (log-normally distributed length)
+        assert not (
+            ak.random_strings_lognormal(2, 1, prob_size)
+            == ak.random_strings_lognormal(2, 1, prob_size)
+        ).all()
+        assert (
+            ak.random_strings_lognormal(2, 1, prob_size, seed=seed).to_list()
+            == ak.random_strings_lognormal(2, 1, prob_size, seed=seed).to_list()
+        )
+
+    @pytest.mark.parametrize("cast_to", CAST_TYPES)
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    def test_cast(self, prob_size, cast_to):
+        arrays = {
+            ak.int64: ak.randint(-(2 ** 48), 2 ** 48, prob_size),
+            ak.float64: ak.randint(0, 1, prob_size, dtype=ak.float64),
+            ak.bool: ak.randint(0, 2, prob_size, dtype=ak.bool),
+        }
+
+        for t1, orig in arrays.items():
+            other = ak.cast(orig, cast_to)
+            assert orig.size == other.size
```

Comment:
once we update `CAST_TYPES` to include `bigint`, we'll need to add back in something like this check
```suggestion
        for t1, orig in arrays.items():
            if t1 == ak.float64 and cast_to == ak.bigint:
                # we don't support casting a float to a bigint
                continue
            other = ak.cast(orig, cast_to)
            assert orig.size == other.size
```

Reply:
Added the check